packages/@react-aria/interactions/test/usePress.test.js

```diff
@@ -2992,4 +2993,161 @@ describe('usePress', function () {
       });
     });
   });
+
+  describe('Owner document and window', () => {
+    installPointerEvent();
+
+    let iframe;
+    let iframeRoot;
+    beforeEach(() => {
+      iframe = document.createElement('iframe');
+      window.document.body.appendChild(iframe);
+      iframeRoot = iframe.contentWindow.document.createElement('div');
+      iframe.contentWindow.document.body.appendChild(iframeRoot);
+    });
+
+    afterEach(() => {
+      iframe.remove();
+    });
+
+    const IframeExample = (props) => {
+      React.useEffect(() => {
+        ReactDOMRender(<Example
+          {...props}
+          data-testid="example" />, iframeRoot);
+      }, [props]);
+
+      return null;
+    };
+
+    test('should handle press events', async () => {
+      let events = [];
+      let addEvent = (e) => events.push(e);
+      render(
+        <IframeExample
+          onPressStart={addEvent}
+          onPressEnd={addEvent}
+          onPressChange={pressed => addEvent({type: 'presschange', pressed})}
+          onPress={addEvent}
+          onPressUp={addEvent} />
+      );
+
+      await waitFor(() => {
+        expect(document.querySelector('iframe').contentWindow.document.body.querySelector('div[data-testid="example"]')).toBeTruthy();
+      });
+
+      const el = document.querySelector('iframe').contentWindow.document.body.querySelector('div[data-testid="example"]');
+      fireEvent(el, pointerEvent('pointerdown', {pointerId: 1, pointerType: 'mouse'}));
+      fireEvent(el, pointerEvent('pointerup', {pointerId: 1, pointerType: 'mouse', clientX: 0, clientY: 0}));
+
+      expect(events).toEqual([
+        {
+          type: 'pressstart',
+          target: el,
+          pointerType: 'mouse',
+          ctrlKey: false,
+          metaKey: false,
+          shiftKey: false,
+          altKey: false
+        },
+        {
+          type: 'presschange',
+          pressed: true
+        },
+        {
+          type: 'pressup',
+          target: el,
+          pointerType: 'mouse',
+          ctrlKey: false,
+          metaKey: false,
+          shiftKey: false,
+          altKey: false
+        },
+        {
+          type: 'pressend',
+          target: el,
+          pointerType: 'mouse',
+          ctrlKey: false,
+          metaKey: false,
+          shiftKey: false,
+          altKey: false
+        },
+        {
+          type: 'presschange',
+          pressed: false
+        },
+        {
+          type: 'press',
+          target: el,
+          pointerType: 'mouse',
+          ctrlKey: false,
+          metaKey: false,
+          shiftKey: false,
+          altKey: false
+        }
+      ]);
+    });
+
+    test('should handle click events and ignore virtual point events', async () => {
```

Comment:
I am not sure this test is helping, if I revert your changes, this one still passes. This is surprising to me. Do you know why that would be?

also, the click event is still fired in the non-virtual case as well, see https://codesandbox.io/s/musing-aryabhata-nhlwyf?file=/src/App.js to see what events the browser fires

I do love adding a test for virtual cursors in addition to pointer. We'll also need one for keyboard I think. Or if it's left off intentionally, that'd be good to know.

Reply:
I believe it's because there's no listener for the click event but only an onClick handler function so that is triggered since we always target the element. 

I'll add a test for the keyboard event.

Reply:
@snowystinger Added the test for keyboard event, hoping this is correct?

Reply:
Sorry, I don't see the keyboard event test. Just remember, the tests you add will help us to not break your code in the future. So even if you've traced the events and have determined you only need to fireEvent with a couple of them, it's better to fire as many as is reasonable. We may change the internals of usePress in the future and depend on those events. We'll try our best to point out those cases where it's better to add an extra seemingly unneeded event.

Reply:
Apologies, added them, lmk if there are other events I should be adding!